ballerina/yaml_api.bal

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Converts YAML string, byte[] or byte-block-stream to subtype of anydata.
+#
+# + s - Source YAML string value or byte[] or byte-block-stream
+# + options - Options to be used for filtering in the projection
+# + t - Target type
+# + return - On success, returns the given target type value, else returns an `yaml:Error`
+public isolated function fromYamlStringWithType(string|byte[]|stream<byte[], error?> s,
+        Options options = {}, typedesc<anydata> t = <>)
+        returns t|Error = @java:Method {'class: "io.ballerina.stdlib.data.yaml.Native"} external;
+        
+# Represents the YAML schema available for the parser.
+#
+# + FAILSAFE_SCHEMA - Generic schema that works for any YAML document
+# + JSON_SCHEMA - Schema supports all the basic JSON types
+# + CORE_SCHEMA - An extension of JSON schema that allows more human-readable presentation
+public enum YAMLSchema {
+    FAILSAFE_SCHEMA,
```

Comment:
```suggestion
    FAIL_SAFE_SCHEMA,
```
wdyt?

Reply:
Will update in the next PR. This PR send to add the workflow and skeleton of the workflow.